zeebe/protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/authorization/AuthorizationRecord.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.protocol.impl.record.value.authorization;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.bufferAsString;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import io.camunda.zeebe.msgpack.property.ArrayProperty;
+import io.camunda.zeebe.msgpack.property.LongProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.msgpack.value.StringValue;
+import io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.camunda.zeebe.protocol.record.value.AuthorizationRecordValue;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import org.agrona.DirectBuffer;
+
+public final class AuthorizationRecord extends UnifiedRecordValue
+    implements AuthorizationRecordValue {
+  private final LongProperty authorizationKeyProp = new LongProperty("authorizationKey", -1L);
+  private final StringProperty ownerTypeProp = new StringProperty("ownerType");
+  private final StringProperty ownerKeyProp = new StringProperty("ownerKey");
+  private final StringProperty resourceKeyProp = new StringProperty("resourceKey");
+  private final StringProperty resourceTypeProp = new StringProperty("resourceType");
```

Comment:
❓ I wonder if this could also be an enum? But I don't know what this would be filled with realistically.

Reply:
I think it could be an enum, essentially it will be a defined list of the possible resources you can apply authorizations to in Camunda, for now I'd leave it as is but we can follow up is thats a better approach